### DISCUSSION TITLE
theorem environments containing just a list

### DISCUSSION THREAD

**Discussion by zeileis** [quarto-dev/quarto-cli#2844] [Q&A]
In quarto 1.1251, when using one of the [theorem divs](https://quarto.org/docs/authoring/cross-references.html#theorems-and-proofs) with just a list (itemized or enumerated), then in HTML output the theorem title becomes the first list element. Only if there is a paragraph prior to the list, it is handled correctly. Example:

```
::: {#def-twothings-plain}
## Plain list
1. First things first.
2. Second thoughts.
:::

::: {#def-twothings-plus}
## List plus intro paragraph
This is about two things.

1. First things first.
2. Second thoughts.
:::
```

Rendered to HTML:

![](https://i.imgur.com/oqY4gJ2.png)

Rendered to PDF it is a bit better but also not ideal:

![](https://i.imgur.com/nmGFrfc.png)

I guess this may be related to the earlier discussion https://github.com/quarto-dev/quarto-cli/discussions/1925 and corresponding fix https://github.com/quarto-dev/quarto-cli/issues/1926.

One can work around the problem by including something like a non-breaking space only in the first paragraph but it would be nice to have having to do this.

**Comment by cscheid** [comment]
Can you try the prerelease 1.2 versions and see if the problem persists?

**Comment by zeileis** [comment]
Thanks for the pointer, I should have tried that myself before. I now tried it with quarto 1.2.215 and indeed there is some progress. The main bug has been fixed:

* With `quarto render --to=html` I get the desired output and the problem is avoided.
* With `quarto render --to=pdf` I still get the output shown above, i.e., the first item is shown on the same line as the definition title.
* With `quarto preview` I somehow cannot switch to a different page. E.g. even if I click on `intro.html` (or explicitly enter that in the browser) I'm somehow always directed to `index.html`. This seems to be completely unrelated to the list issue, though.

**Reply by cscheid** [reply]
Thanks, I can repro the PDF problem as well. I'll create an issue to track.
